tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java

```diff
@@ -251,4 +251,26 @@ public void testTonyResourcesFlag() throws ParseException {
     int exitCode = client.start();
     Assert.assertEquals(exitCode, 0);
   }
+
+
+  /*
+   * In conditional_wait.py script, we wait for 5s for chief job type. If we only wait for worker to finish, this test
+   * is gonna fail.
```

Comment:
I don't understand why this test would fail even if we only waited for non-chief workers to finish.

Reply:
This test tries to mimic the scenario happens in #134. 

Prior to this change, we don't wait for non-chief worker to finish and conclude the training. The logic in TonySession's `updateSessionStatus()` method used to make decision based on if non-ps worker has finished, if no -> setFinalStatus false.

Reply:
Got it, that makes sense. Thanks for explaining.